src/test/java/org/folio/rest/impl/GOBIIntegrationServiceResourceImplTest.java

```diff
@@ -614,6 +621,223 @@ public final void testPostGobiOrdersFallBackLocation(TestContext context) throws
+
+    logger.info("End: Testing for 201 - posted order returns existing Order if present");
+  }
+
+  @Test
+  public final void testPostGobiOrdersFailedToRetrieveExistingOrder() throws Exception {
+    logger.info("Begin: Testing for 201 - Create new Order if retrieving existing Order fails");
+
+
+    final String body = getMockData(POLISTEDELECTRONICSERIALPATH);
+
+    final GobiResponse order = RestAssured
+      .given()
+        .header(TOKENHEADER)
+        .header(URLHEADER)
+        .header(TENANTHEADER)
+        .header(CONTENTTYPEHEADERXML)
+        .header(new Header(MOCK_OKAPI_GET_ORDER_HEADER, MOCK_INSTRUCTION_FAIL_ORDER))
+        .body(body)
+      .when()
+        .post(ORDERSPATH)
+      .then()
+        .statusCode(201)
+        .contentType(ContentType.XML)
+        .extract()
+          .body()
+            .as(GobiResponse.class, ObjectMapperType.JAXB);
+
+
+    //should try to fetch the order, and gets 500
+    List<JsonObject> getOrder = MockServer.serverRqRs.get(PURCHASEORDER, HttpMethod.GET);
+    assertNull(getOrder);
+
+    //should create an order if the call to fetch existing order fails
+    List<JsonObject> postedOrder = MockServer.serverRqRs.get(PURCHASEORDER, HttpMethod.POST);
+    assertEquals(1, postedOrder.size());
+
+    //should not call PUT to open order
+    List<JsonObject> putOrder = MockServer.serverRqRs.get(COMPOSITE_PURCHASE_ORDER, HttpMethod.PUT);
+    assertNull(putOrder);
+
+    //return the created PO Line Number
+    assertNotNull(order.getPoLineNumber());
+
+    logger.info("End: Testing for 201 - Create new Order if retrieving existing Order fails");
+  }
+
+  @Test
+  public final void testPostGobiOrdersRetryToOpenOrderFails() throws Exception {
+    logger.info("Begin: Testing for 201 - Return existing Order even if it is pending, and retry to Open Fails");
+
+    final String body = getMockData(POLISTEDELECTRONICSERIALPATH);
+
+    final GobiResponse order = RestAssured
+      .given()
+        .header(TOKENHEADER)
+        .header(URLHEADER)
+        .header(TENANTHEADER)
+        .header(CONTENTTYPEHEADERXML)
+        .header(new Header(MOCK_OKAPI_GET_ORDER_HEADER, MOCK_INSTRUCTION_GET_PENDING_ORDER))
+        .header(new Header(MOCK_OKAPI_PUT_ORDER_HEADER, MOCK_INSTRUCTION_PUT_FAIL))
+        .body(body)
+      .when()
+        .post(ORDERSPATH)
+      .then()
+        .statusCode(201)
+        .contentType(ContentType.XML)
+        .extract()
+          .body()
+            .as(GobiResponse.class, ObjectMapperType.JAXB);
+
+    //should fetch existing order which is in Pending state
+    List<JsonObject> getOrder = MockServer.serverRqRs.get(PURCHASEORDER, HttpMethod.GET);
+    assertEquals(1, getOrder.size());
+
+    //should try to Open, the existing Pending Order, which fails
+    List<JsonObject> putOrder = MockServer.serverRqRs.get(COMPOSITE_PURCHASE_ORDER, HttpMethod.PUT);
+    assertEquals(1, putOrder.size());
+
+    //should get the existing Order Line Number
```

Comment:
This is only getting an existing pending order so maybe description should say that.

Reply:
the line 750 has the required comments as it is the place , however added the comment here too